src/main/java/teammates/ui/webapi/Action.java

```diff
@@ -111,6 +114,23 @@ public void checkAccessControl() throws UnauthorizedAccessException {
         checkSpecificAccessControl();
     }
 
+    /**
+     * Gets the user information of the current user.
+     */
+    public Map<String, String> getUserInfoForLogging() {
+        Map<String, String> info = new HashMap<>();
+
+        String googleId = userInfo == null ? null : userInfo.getId();
+
+        info.put("googleId", googleId);
+        if (userInfoForUserWithNoAccount.isEmpty()) {
+            info.put("regkey", getRequestParamValue(Const.ParamsNames.REGKEY));
```

Comment:
This line is to account for the brief time period that the instructor operates with regkey.